File: frontend/src/app/models/monster/monster.ts
import {MovementSpeed} from "./movement-speed";
import {StatBlock} from "./stat-block";
import {Characteristic} from "./characteristic";
import {Action} from "./action";
import {Alignment} from "./alignment";
import {ArmorClass} from "./armor-class";
import {Skill} from "./skill";

export class Monster {
  uuid?: string;
  name?: string;
  size?: string;
  type?: string;
  alignment?: Alignment;
  armorClass?: ArmorClass;
  hitPoints?: number;
  movementSpeed?: MovementSpeed;
  attributes?: StatBlock;
  savingThrowModifiers?: StatBlock;
  skills?: Skill[];
  damageImmunities?: string[];
  senses?: string[];
  languages?: string[];
  challengeRating?: number;
  characteristics?: Characteristic[];
  actions?: Action[];
  monsterDescription?: string;
  imageIds?: string[]
}

File: src/main/java/de/urr4/monsters/domain/monster/Monster.java
package de.urr4.monsters.domain.monster;

import lombok.*;

import java.util.List;
import java.util.UUID;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class Monster {
    private UUID uuid;
    private String name;
    private Size size;
    private String type;
    private Alignment alignment;
    @With
    private ArmorClass armorClass;
    private int hitPoints;
    private MovementSpeed movementSpeed;
    private StatBlock<Integer> attributes;
    @With
    private StatBlock<String> savingThrowModifiers;
    private List<Skill> skills;
    private List<String> damageImmunities;
    private List<String> senses;
    private List<String> languages;
    private int challengeRating;
    private List<Characteristic> characteristics;
    private List<Action> actions;

    private String monsterDescription;

    private List<UUID> imageIds;
}
